client/me/account/main.jsx

```diff
@@ -492,7 +502,8 @@ const Account = React.createClass( {
 						id="url"
 						name="url"
 						onFocus={ this.recordFocusEvent( 'Web Address Field' ) }
-						valueLink={ this.valueLink( 'user_URL' ) }
+						value={ this.getUserSetting( 'user_URL' ) || '' }
+						onChange={ this.updateUserSettingInput( 'user_URL' ) }
```

Comment:
just a note on this here related to the previous discussion in https://github.com/Automattic/wp-calypso/pull/12729#issuecomment-291795046

moving the closure generator into a separate function doesn't actually do anything different than inlining the arrow expression inside of the JSX. so while we have made the linter happy by extracting it we haven't actually addressed the reason that linting is there in the first place

when we need to respond to events and pass data into that handler then the ideal situation is simply to leave the information in the JSX and _stop_ creating a new handler on every call to `render()`

```js
updateUserSetting = event => {
	const value = event.target.value;
	const setting = event.target.dataset.settingName

	update( setting, value );
}

<FormTextInput
	…
	data-setting-name="user_URL"
	onChange={ this.updateUserSetting }
/>
```

note the use of the custom data attributes to hold the additional properties